channel_test.go

```diff
@@ -112,3 +113,18 @@ func TestIsolatedSubChannelsDontSharePeers(t *testing.T) {
 	assert.NotNil(t, sub.peers.peersByHostPort["127.0.0.1:3000"])
 	assert.Nil(t, isolatedSub.peers.peersByHostPort["127.0.0.1:3000"])
 }
+
+func TestChannelTracerMethod(t *testing.T) {
+	tracer := mocktracer.New()
+	ch, err := NewChannel("svc", &ChannelOptions{
+		Tracer: tracer,
+	})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.Equal(t, tracer, ch.Tracer(), "expecting tracer passed at initialization")
+
+	ch, err = NewChannel("svc", &ChannelOptions{})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.EqualValues(t, opentracing.GlobalTracer(), ch.Tracer(), "expecting default tracer")
```

Comment:
I see what you mean, but my concern with that is that there is actually a global variable behind GlobalTracer(), and changing a global variable in a unit test feels iffy. Probably no harm, but if any tests happen to run concurrently, it _could_ be a problem. Wdyt?